core/src/main/java/org/infinispan/commands/ReplicableCommand.java

```diff
@@ -19,34 +19,46 @@
  */
 public interface ReplicableCommand {
    /**
-    * Performs the primary function of the command.  Please see specific implementation classes for details on what is
-    * performed as well as return types. <b>Important</b>: this method will be invoked at the end of interceptors chain.
-    * It should never be called directly from a custom interceptor.
+    * Invoke the command asynchronously.
     *
-    * @param ctx invocation context
-    * @return arbitrary return value generated by performing this command
-    * @throws Throwable in the event of problems.
-    * @deprecated Since 9.0, only {@link VisitableCommand}s should implement {@code perform(InvocationContext)}.
+    * <p>This method replaces {@link #perform(InvocationContext)} for remote execution.
+    * The default implementation and {@link #perform(InvocationContext)} will be removed in future versions.
+    * </p>
+    *
+    * @since 9.0
     */
-   @Deprecated
-   default Object perform(InvocationContext ignored) throws Throwable {
-      return invoke();
+   default CompletableFuture<Object> invokeAsync() throws Throwable {
+      return CompletableFuture.completedFuture(perform(null));
```

Comment:
I changed the order to put `invokeAsync` first, because that's the one I want commands to actually implement :)

I don't think I can preserve compatibility with existing custom commands and also prepare for removing `perform(InvocationContext)` without making every method `default`.